madmom/audio/signal.py

```diff
@@ -1364,21 +1374,22 @@ def process(self, data, **kwargs):
         ----------
         data : :class:`Signal` instance
             Signal to be sliced into frames.
-        kwargs : dict
-            Keyword arguments passed to :class:`FramedSignal` to instantiate
-            the returned object.
+        kwargs : dict, optional
+            Keyword arguments passed to :class:`FramedSignal`.
 
         Returns
         -------
         frames : :class:`FramedSignal` instance
             FramedSignal instance
 
         """
+        # update arguments passed to FramedSignal
+        args = dict(frame_size=self.frame_size, hop_size=self.hop_size,
+                    fps=self.fps, origin=self.origin, end=self.end,
+                    num_frames=self.num_frames)
+        args.update(kwargs)
```

Comment:
Is there a reason for this change? I actually think the previous code was more direct and thus easier understandable.